src/dbnode/persist/fs/seek_manager.go

```diff
@@ -795,7 +836,7 @@ func (m *seekerManager) seekersByTime(shard uint32) *seekersByTime {
 	// Check if raced with another call to this method
 	if int(shard) < len(m.seekersByShardIdx) {
 		byTime := m.seekersByShardIdx[shard]
-		return byTime
+		return byTime, true
```

Comment:
After re-acquiring the write lock we need to check `if !m.shardExistsWithLock(shard) {` again here too I believe (since could change after re-acquiring the lock).